src/libs/actions/Policy.js

```diff
@@ -57,6 +57,26 @@ function getSimplifiedEmployeeList(employeeList) {
     return employeeListEmails;
 }
 
+/**
+ * Used to update ALL of the policies at once. If a policy is present locally, but not in the policies object passed here it will be removed.
+ * @param {Object} policyCollection - object of policy key and partial policy object
+ */
+function updateAllPolicies(policyCollection) {
+    // Clear out locally cached policies that have been deleted (i.e. they exist locally but not in our new policy collection object)
+    _.each(allPolicies, (policy, key) => {
+        if (policyCollection[key]) {
+            return;
+        }
+
+        Onyx.set(key, null);
+    });
+
+    // Set all the policies
+    _.each(policyCollection, (policyData, key) => {
+        Onyx.merge(key, policyData);
```

Comment:
Do I understand it correctly that `Onyx.merge(key, policyData)` is inserting entries into the `allPolicies` object?

If that is the case, why bother iterating through all of the entries in `policyCollection` and `allPolicies` when you can just set `allPolicies` equal to `policyCollection`.

```jsx
function updateAllPolicies(policyCollection) {
    // Make the locally cached policies the same as the new policy collection object.
    allPolicies = policyCollection;
}
```

Reply:
>Do I understand it correctly that Onyx.merge(key, policyData) is inserting entries into the allPolicies object?

More or less but I'd say that the data is being updated and then subscribers are notified of the change. See where `allPolicies` gets updated by the subscriber here: 

https://github.com/Expensify/App/blob/0a77eb2e3e8cc399c8571bc7a223963ad300fbdf/src/libs/actions/Policy.js#L14-L22

Once we merge a collection key a subscriber to that collection will update and do this for us - so there's no reason to do `allPolicies = policyCollection`. We _could_ do this and my expectation would be that the end result is the same - but I can't think of why we would.

>why bother iterating through all of the entries in policyCollection and allPolicies when you can just set allPolicies equal to policyCollection

We're doing this so that the data gets persisted in device storage (that's what Onyx does). What you are suggesting would work if we were only dealing with a local variables in memory, but we're doing something else. [Check this out](https://github.com/Expensify/App#philosophy) and lmk if you still have any questions!